### PR TITLE
fix(route53): use static s3 website endpoint

### LINKED ISSUE #4248: Route 53: Target S3 Alias shouldn't target a bucket
The current format of Route53 `BucketWebsiteTarget` is incorrect. It requires the following two rules to be followed:

* The `DnsName` property is set depending on the region, but not the bucket name
* The website Bucket name must match the full record and domain name

The [Knowledge Center](https://aws.amazon.com/premiumsupport/knowledge-center/route-53-no-targets/) makes the last rule explicit:

> For example, if your bucket's name is bucket.example.com, the record name must also be bucket.example.com.

### Reproduction Steps

The following stack will deploy, but won't behave correctly because it breaches the first rule:

```ts
const [recordName, domainName] = ['foo', 'example.com'];
const bucketName = `${recordName}.${domainName}`;

const bucketWebsite = new Bucket(this, 'BucketWebsite', {
  bucketName,
  publicReadAccess: true,
  websiteIndexDocument: 'index.html',
});

const zone = HostedZone.fromLookup(this, 'Zone', {domainName});

new ARecord(this, 'AliasBucket', {
  zone,
  recordName,
  // Currently, DnsName is `${bucketName}.s3-website.${this.region}.amazonaws.com`
  target: RecordTarget.fromAlias(new BucketWebsiteTarget(bucketWebsite)),
});
```

### Error Log

The previous stack deploys, but the domain name cannot be resolved:

```sh
> curl -vv foo.example.com
* Rebuilt URL to: foo.example.com/
* Could not resolve host: foo.example.com
* Closing connection 0
curl: (6) Could not resolve host: foo.example.com
```

### Environment

  - **CLI Version      :** 1.9.0
  - **Framework Version:** 1.9.0
  - **OS               :** all
  - **Language         :** all

### Other

See [#3928](https://github.com/aws/aws-cdk/issues/3928#issuecomment-535221648) for @jokyjang's original report

#### Temporary fix for `v1.9.0`

This stack deploys and behaves correctly, overriding the `bucketWebsiteDomainName` set by the CDK:

```ts
// ...

new ARecord(this, 'AliasBucket', {
  zone,
  recordName,
  // @ts-ignore
  target: RecordTarget.fromAlias(new BucketWebsiteTarget({
    ...bucketWebsite,
    bucketWebsiteDomainName: `s3-website.${this.region}.amazonaws.com`,
  })),
});
```

#### `CNAME` record

From what I've seen, it's also not possible to create a direct `CNAME` for that `A` Record. Trying to resolve a CNAME pointing to a Bucket Alias record causes the same `Could not resolve host: bar.example.com` error. 
The [Route 53 developer guide](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/getting-started.html#getting-started-create-s3-www-bucket) hints at creating another `bar.example.com` bucket, and redirecting it to the original `foo.example.com`

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by nmussy**
`DnsName` shouldn't target the bucket website domain name.
Instead, `DnsName` must point to the region-wide S3 endpoint,
and the bucket name must match the full DNS record

* Replace `bucketWebsiteDomainName` with `s3StaticWebsiteEndpoint`
* Document "Bucket name must match the full DNS name"

Fixes #4248

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-route53-targets/README.md [hunk 1]
```diff
@@ -48,5 +41,26 @@ This library contains Route53 Alias Record targets for:
     // or - route53.RecordTarget.fromAlias(new alias.ApiGatewayDomainName(domainName)),
   });
   ```
+* S3 Bucket WebSite:
+
+**Important:** The Bucket name must strictly match the full DNS name.
```
Comment (nmussy) [inline packages/@aws-cdk/aws-route53-targets/README.md]: I'd like to further improve the user experience and check in `aws-route53` that the Bucket name matches `RecordSet` and `HostedZone`. How do you guys feel about that?

Comment (eladb) [reply packages/@aws-cdk/aws-route53-targets/README.md]: I like it!

Comment (nmussy) [reply packages/@aws-cdk/aws-route53-targets/README.md]: I'll make another PR for it 👍

**Comment by nmussy** [comment]
Thanks for the quick response @eladb 👍

**Pull request merged**